[Weave] Reference dataset when using eval logger (#2288)
wandb/docs: weave/guides/evaluation/evaluation_logger.mdx

### Link to an existing dataset

> When you pass raw dicts as `inputs` to `log_prediction`, Weave re-ingests the data with every evaluation run. For large datasets, this is inefficient because the same data is stored multiple times.

Reviewer: Is "raw dict" really what they are called? What about something like "This stores duplicate data, which may waste space if the dataset is large or if a large number of evaluations reuse it" or something?

> To avoid this, publish your dataset to Weave first, then pass the published dataset's rows as `inputs`. Published rows carry internal references that Weave resolves instead of re-ingesting. This gives you the same linked experience as the standard [Evaluation framework](../core-types/evaluations), where each prediction links back to a specific dataset row in the Weave UI.

Reviewer: ```suggestion
To avoid this duplication, publish your dataset to Weave before running any evaluations, then pass the published dataset's rows as `inputs`. Weave resolves references to published rows using internal references instead of re-ingesting the data. This technique gives you the same linked experience as the standard [Evaluation framework](../core-types/evaluations), where each prediction links back to a specific dataset row in the Weave UI.
```
Something like this? Ignore if I'm getting it wrong.

> The following example shows how to publish a dataset and link to it in the `EvaluationLogger`. Once you have retrieved the dataset, you can access and iterate over it like any other dataset.

Reviewer: ```suggestion
The following example publishes a dataset and links to it in the `EvaluationLogger`, before retrieving and iterating over it like any other dataset.
```
